### PR TITLE
Introduce Recorded Tests and Samples to Media Composition

### PR DESCRIPTION
**Pull request by minnieliu**
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

**Review by alexokun — APPROVED** [review]

File: sdk/communication/Azure.Communication.MediaComposition/tests/MediaCompositionClient/MediaCompositionClientLiveTests.cs [hunk 1]
```diff
@@ -0,0 +1,391 @@
...
+                            GroupCall = new("d12d2277-ffec-4e22-9979-8c0d8c13d193")
+                        }
+                    }
+                };
+                await mediaCompositionClient.CreateAsync(mediaCompositionId, layout, null, outputs);
+            }
+            catch (RequestFailedException ex)
+            {
+                Assert.NotNull(ex.Message);
+                Assert.True(ex.Message.Contains("400"));
```
Comment (DominikMe) [inline sdk/communication/Azure.Communication.MediaComposition/tests/MediaCompositionClient/MediaCompositionClientLiveTests.cs:90]: Should we check the Message or rather the Status here? I would imagine that the message will change to something more specific in the future.

File: sdk/communication/Azure.Communication.MediaComposition/tests/MediaCompositionClient/MediaCompositionClientLiveTests.cs [hunk 2]
```diff
@@ -0,0 +1,391 @@
...
+                    Resolution = new(1920, 1080),
+                    Presenter = new("presenter", "support")
+                };
+
+                await mediaCompositionClient.CreateAsync(mediaCompositionId, layout, presentationInputs, null);
+            }
+            catch (RequestFailedException ex)
+            {
+                Assert.NotNull(ex.Message);
+                Assert.True(ex.Message.Contains("400"));
```
Comment (DominikMe) [inline sdk/communication/Azure.Communication.MediaComposition/tests/MediaCompositionClient/MediaCompositionClientLiveTests.cs:114]: Same comment about asserting the status instead.

File: sdk/communication/Azure.Communication.MediaComposition/tests/MediaCompositionClient/MediaCompositionClientLiveTests.cs [hunk 3]
```diff
@@ -0,0 +1,391 @@
...
+                            GroupCall = new("d12d2277-ffec-4e22-9979-8c0d8c13d193")
+                        }
+                    }
+                };
+                await mediaCompositionClient.CreateAsync(mediaCompositionId, layout, inputs, outputs);
+            }
+            catch (RequestFailedException ex)
+            {
+                Assert.NotNull(ex.Message);
+                Assert.True(ex.Message.Contains("400"));
```
Comment (DominikMe) [inline sdk/communication/Azure.Communication.MediaComposition/tests/MediaCompositionClient/MediaCompositionClientLiveTests.cs:158]: Status, see above

Comment (DominikMe) [reply sdk/communication/Azure.Communication.MediaComposition/tests/MediaCompositionClient/MediaCompositionClientLiveTests.cs:158]: And for all other occurrences :-)

File: sdk/communication/Azure.Communication.MediaComposition/tests/MediaCompositionClient/MediaCompositionClientLiveTests.cs [hunk 4]
```diff
@@ -0,0 +1,391 @@
...
+            }
+            catch (RequestFailedException ex)
+            {
+                Assert.NotNull(ex.Message);
+                Assert.True(ex.Message.Contains("404"));
+                Assert.True(ex.Message.Contains("Not Found"));
+                Console.WriteLine(ex.Message);
+                return;
+            }
+            Assert.Fail("An exception should have been thrown trying to start a non-existent media composition..");
```
Comment (DominikMe) [inline sdk/communication/Azure.Communication.MediaComposition/tests/MediaCompositionClient/MediaCompositionClientLiveTests.cs:252]: nit: one `.` too many at the end

**Review by 0rland0Wats0n — APPROVED** [review]
LGTM

**Review by DominikMe — APPROVED** [review]
Great job! Lgtm with minor comments

File: sdk/communication/Azure.Communication.MediaComposition/README.md [hunk 5]
```diff
@@ -48,23 +48,13 @@ var client = new MediaCompositionClient(endpoint, tokenCredential);
 ### Creating the media composition
```
[the remark this answers is not in the document]
Comment (minnieliu) [reply sdk/communication/Azure.Communication.MediaComposition/README.md:48]: I think we might be better off doing this when we have deployed to prod. We currently only have Dev and Int, which can be unstable

File: sdk/communication/Azure.Communication.MediaComposition/tests/MediaCompositionClient/MediaCompositionClientLiveTests.cs (hunk 1 above)
Comment (minnieliu) [reply sdk/communication/Azure.Communication.MediaComposition/tests/MediaCompositionClient/MediaCompositionClientLiveTests.cs:90]: Great suggestion! Will update in my next commit

File: sdk/communication/Azure.Communication.MediaComposition/tests/MediaCompositionClient/MediaCompositionClientLiveTests.cs [hunk 6]
```diff
@@ -0,0 +1,391 @@
...
+using Azure.Communication.MediaComposition.Models;
+using Azure.Communication.Tests;
+using NUnit.Framework;
+
+namespace Azure.Communication.MediaComposition.Tests
+{
+    public class MediaCompositionClientLiveTests : MediaCompositionClientLiveTestBase
+    {
+        private const string mediaCompositionId = "presentation";
+        private readonly Dictionary<string, MediaInput> presentationInputs = new Dictionary<string, MediaInput>()
```
[the remark this answers is not in the document]
Comment (minnieliu) [reply sdk/communication/Azure.Communication.MediaComposition/tests/MediaCompositionClient/MediaCompositionClientLiveTests.cs]: Done, thank you!

**Pull request merged**